### PR TITLE
Modeling problem

### PR DESCRIPTION
**Pull request by trkaplan**
Need help with sequelize. 
Getting this error.

```bash
== 20181230220357-events: migrating =======
ERROR: insert or update on table "events" violates foreign key constraint "events_uuid_fkey"
```

### CODE REVIEW

File: src/models/user.js [hunk 1]
```diff
@@ -91,7 +91,9 @@ module.exports = (sequelize: Sequelize, dataTypes: DataTypes) => {
   });
 
   User.associate = (models) => {
+    User.hasMany(models.UserEvent);
     User.belongsToMany(models.Community, { through: models.CommunityUser });
+    User.belongsToMany(models.Event, { through: models.UserEvent, foreignKey: 'uuid' });
```
Comment (SelmanKahya) [inline src/models/user.js]: sanirim foreign key e gerek yok, varsa da userUuid olmali, degil mi?

File: scripts/db/development/seeders/20181230234701-users-events.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+const items = [
+  {
+    event_uuid: '3346776a-d69d-11e8-9f8b-f2801f1b9fd1',
```
Comment (SelmanKahya) [inline scripts/db/development/seeders/20181230234701-users-events.js]: bu dosyadaki tum event ve user uuid ler gercekten users ve events tablolarinda mevcut mu?

File: scripts/db/development/seeders/20181230220357-events.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    time_start: new Date(),
+    time_end: new Date(),
+    time_zone: 'Turkey Standard Time',
+    location: 'online',
+    created_at: new Date(),
+    updated_at: new Date(),
+  },
+  {
+    uuid: '3346776a-d69d-11e8-9f8b-f2801f1b9fc5',
+    community_uuid: '6c52dc16-d6a6-11e8-9f8b-f2801f1b9fd1',
```
Comment (SelmanKahya) [inline scripts/db/development/seeders/20181230220357-events.js]: bu comminty uuid ler community tablosunda mevcut mu? mobildeyim bakamoyorum

File: src/models/community.js [hunk 4]
```diff
@@ -91,6 +91,7 @@ module.exports = (sequelize: Sequelize, dataTypes: DataTypes) => {
     Community.hasMany(models.ConversationCategory, {
       foreignKey: 'communityUuid',
     });
+    Community.hasMany(models.Event, { foreignKey: 'uuid', targetKey: 'uuid' });
```
Comment (trkaplan) [inline src/models/community.js]: This relation was causing the error.
Fixed this one by replacing with:
`Community.hasMany(models.Event, { foreignKey: 'communityUuid', targetKey: 'uuid' });`

**Comment by trkaplan** [comment]
Now getting this error:
```bash
== 20181230234701-users-events: migrating =======
ERROR: column "event_uuid" of relation "events" does not exist
```

File: src/models/user.js (hunk 1 above)
Comment (trkaplan) [reply src/models/user.js]: UserEvent'deki adı evet `uuid` yerine, `userUuid`olmalı fakat bunu denediğimde de hatayı almaya devam ediyorum.

**Comment by trkaplan** [comment]
I tried different combinations, read documentation, watched some screencasts, asked for help to the Kommunity Discord channel, no luck.

For now, I’m using a mock json instead of data from database.


Selman Kahya <notifications@github.com> şunları yazdı (9 Oca 2019 09:23):

**Comment by SelmanKahya** [comment]
can you talk to @bariscc? he just submitted a diff with relations.

**Comment by SelmanKahya** [comment]
:(

**Comment by SelmanKahya** [comment]
I will merge and try to address the issues

**Pull request merged**